tests/Fixer/StringNotation/MultilineStringToHeredocFixerTest.php

```diff
@@ -0,0 +1,362 @@
+        ];
+
+        yield 'colliding closing marker - two' => [
+            <<<'EOF'
+                <?php
+                $a = <<<'EOD__'
+                line1
+                EOD
+                EOD_
+                line2
+                EOD__;
+                EOF,
+            <<<'EOF'
+                <?php
+                $a = 'line1
+                EOD
+                EOD_
+                line2';
+                EOF,
+        ];
+
+        yield 'single quoted unescape' => [
+            <<<'EOF'
+                <?php
+                $a = <<<'EOD'
+                line1
+                \
+                \n
+                '
+                \\'
+                \"
+                \
+
+                EOD;
+                EOF,
+            <<<'EOD'
+                <?php
+                $a = 'line1
+                \\
+                \n
+                \'
+                \\\\\'
+                \"
+                \
+                ';
+                EOD,
+        ];
+
+        yield 'double quoted unescape' => [
+            <<<'EOF'
+                <?php
+                $a = <<<EOD
+                line1
+                \\
+                \n
+                "
+                \\\\"
+                \'
+                \
+                "{$rawPath}"
+
+                EOD;
+                EOF,
+            <<<'EOD'
+                <?php
+                $a = "line1
+                \\
+                \n
+                \"
+                \\\\\"
+                \'
+                \
+                \"{$rawPath}\"
+                ";
+                EOD,
+        ];
+
+        yield 'single quoted /w variable' => [
+            <<<'EOF'
+                <?php
+                $a = <<<'EOD'
+                line1$var
+                line2
+                EOD;
+                EOF,
+            <<<'EOD'
+                <?php
+                $a = 'line1$var
+                line2';
+                EOD,
+        ];
+
+        yield 'double quoted /w simple variable' => [
+            <<<'EOF'
+                <?php
+                $a = <<<EOD
+                line1$var
+                line2
+                EOD;
+                EOF,
+            <<<'EOD'
+                <?php
+                $a = "line1$var
+                line2";
+                EOD,
+        ];
+
+        yield 'double quoted /w simple curly variable' => [
+            <<<'EOF'
+                <?php
+                $a = <<<EOD
+                line1{$var}
+                line2
+                EOD;
+                EOF,
+            <<<'EOD'
+                <?php
+                $a = "line1{$var}
+                line2";
+                EOD,
+        ];
+
+        yield 'double quoted /w complex curly variable' => [
+            <<<'EOF'
+                <?php
+                $a = <<<EOD
+                {$arr['foo'][3]}
+                { $obj->values[3]->name }
+                {${getName()}}
+                EOD;
```

Comment:
Is it possible to achieve:

```suggestion
                $a = <<<EOD
                    {$arr['foo'][3]}
                    { $obj->values[3]->name }
                    {${getName()}}
                    EOD;
```

From `$a`'s perspective it's the same, but it's more readable.

Reply:
Identation is handled by another fixer.

Reply:
Which one? I took a look at [string notation fixers](https://cs.symfony.com/doc/rules/index.html#string-notation) and did not see such. Would be good to have integration test for it 🙂.

Reply:
`heredoc_indentation`

Reply:
Ah, it's under "Whitespace" category. Thanks for integration test!